File: 2020/01/day01_1.rb
report = []
file = File.new("input.txt", "r")
while (line = file.gets)
  report.push(line.to_i)
end
file.close

report.each do |i|
  report.each do |j|
    if( i + j == 2020)
      puts i*j
      exit
    end
  end
end

File: 2020/01/day01_1.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strconv"
)

func main() {
  f, err := os.Open("input_test.txt")

  if err != nil {
    log.Fatal(err)
  }
  defer f.Close()

  var lines []int
  
  scanner := bufio.NewScanner(f)
  for scanner.Scan() {  	
  	line, err := strconv.Atoi(scanner.Text())		
  	if err != nil {
    	log.Fatal(err)
  	}

  	lines = append(lines, line)
  }


  if err := scanner.Err(); err != nil {
   log.Fatal(err)
  }

	for _, a := range lines {
  	for _, b := range lines {
  		if (a + b == 2020) {
  			fmt.Println(a*b)
  			return
  		}  		
  	}  	
  }


 }